app/src/main/java/agency/five/codebase/android/movieapp/ui/movieDetails/MovieDetailsScreen.kt

```diff
@@ -0,0 +1,257 @@
+            FavoriteButton(
+                modifier = Modifier
+                    .padding(
+                        start = dimensionResource(id = R.dimen.movie_details_hero_movie_title_padding),
+                        bottom = MaterialTheme.spacing.medium
+                    ),
+                isFavorite = movieDetailsViewState.isFavorite,
+                onCLick = {}
+            )
+        }
+    }
+}
+
+@Composable
+fun MovieDetailsOverviewSection(
+    movieDetailsViewState: MovieDetailsViewState,
+    modifier: Modifier = Modifier
+) {
+    Column(
+        modifier = modifier
+            .fillMaxWidth()
+            .padding(
+                start = MaterialTheme.spacing.medium,
+                end = MaterialTheme.spacing.medium
+            )
+    ) {
+        Text(
+            text = stringResource(id = R.string.overview_text),
+            style = HeaderText,
+            maxLines = 1,
+            overflow = TextOverflow.Ellipsis,
+            modifier = Modifier
+                .fillMaxWidth()
+        )
+        Spacer(
+            modifier = Modifier
+                .height(
+                    MaterialTheme.spacing.small
+                )
+        )
+        Text(
+            text = movieDetailsViewState.overview,
+            style = Typography.body1,
+            overflow = TextOverflow.Ellipsis,
+            modifier = Modifier
+                .fillMaxWidth()
+        )
+        Spacer(
+            modifier = Modifier
+                .height(
+                    MaterialTheme.spacing.medium
+                )
+        )
+        LazyHorizontalGrid(
+            rows = GridCells.Fixed(2),
+            horizontalArrangement = Arrangement.SpaceBetween,
+            modifier = Modifier.height(
+                dimensionResource(id = R.dimen.movie_details_crewman_grid_height)
+            )
+                .fillMaxWidth()
+        ) {
+            items(
+                items = movieDetailsViewState.crew,
+                key = { crewman ->
+                    crewman.id
+                }
+            ) { crewman ->
+                CrewItem(
+                    viewState = CrewItemViewState(crewman.name, crewman.job)
+                )
+            }
+        }
+    }
+}
+
+@Composable
+fun MovieDetailsCastSection(
+    movieDetailsViewState: MovieDetailsViewState,
+    modifier: Modifier = Modifier
+) {
+    Column(
+        modifier = modifier
+            .padding(
+                start = MaterialTheme.spacing.medium,
+                bottom = MaterialTheme.spacing.medium
+            )
+    ) {
+        Text(
+            text = stringResource(id = R.string.top_billed_cast_text),
+            style = HeaderText,
+            maxLines = 1,
+            overflow = TextOverflow.Ellipsis,
+            modifier = Modifier
+                .fillMaxWidth()
+        )
+        Spacer(modifier = Modifier.height(MaterialTheme.spacing.medium))
+        LazyRow(
```

Comment:
again view is cut of by padding, should be from end to end with first and last element having padding defined with `contentPadding`